### PR TITLE
Milestoning: Bitemp VLF Exclude Source Validity From Time Column in Sink

### PR DESCRIPTION
**Pull request by kumuwu**
#### What type of PR is this?

Feature fix

#### What does this PR do / why is it needed ?

User wants to exclude the source validity from time column in the sink

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
No

### CODE REVIEW

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java [hunk 1]
```diff
@@ -119,12 +121,13 @@ class BitemporalDeltaPlanner extends BitemporalPlanner
...
+            this.sourceValidDatetimeFrom = FieldValue.builder().fieldName(ingestMode.validityMilestoning().validityDerivation().accept(BitemporalPlanner.EXTRACT_SOURCE_VALID_DATE_TIME_FROM)).alias(VALID_DATE_TIME_FROM_NAME).build();
+            this.targetValidDatetimeFrom = FieldValue.builder().fieldName(ingestMode.validityMilestoning().accept(BitemporalPlanner.EXTRACT_TARGET_VALID_DATE_TIME_FROM)).alias(VALID_DATE_TIME_FROM_NAME).build();
+            this.targetValidDatetimeThru = FieldValue.builder().fieldName(ingestMode.validityMilestoning().accept(BitemporalPlanner.EXTRACT_TARGET_VALID_DATE_TIME_THRU)).alias(VALID_DATE_TIME_THRU_NAME).build();
 
             this.dataFields = stagingDataset().schemaReference().fieldValues().stream().map(field -> FieldValue.builder().fieldName(field.fieldName()).build()).collect(Collectors.toList());
             this.dataFields.removeIf(field -> field.fieldName().equals(ingestMode.digestField()));
+            this.dataFields.removeIf(field -> field.fieldName().equals(ingestMode.validityMilestoning().validityDerivation().accept(BitemporalPlanner.EXTRACT_SOURCE_VALID_DATE_TIME_FROM)));
```
Comment (prasar-ashutosh) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java]: We can rewrite this as:
            this.dataFields.removeIf(field -> field.fieldName().equals(sourceValidDatetimeFrom.fieldName));

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java [hunk 2]
```diff
@@ -436,77 +447,73 @@ private Insert getStageToTemp()
...
+            selectX1 = Selection.builder().source(stagingDataset()).addAllFields(primaryKeyFieldsAndFromFieldFromStage).alias(LEFT_DATASET_IN_JOIN_ALIAS).build();
         }
-        Selection selectY1 = Selection.builder().source(mainDataset()).condition(openRecordCondition).addAllFields(primaryKeyFieldsAndFromField).alias(RIGHT_DATASET_IN_JOIN_ALIAS).build();
+        Selection selectY1 = Selection.builder().source(mainDataset()).condition(openRecordCondition).addAllFields(primaryKeyFieldsAndFromFieldFromMain).alias(RIGHT_DATASET_IN_JOIN_ALIAS).build();
 
         Condition x1AndY1PkMatchCondition = LogicalPlanUtils.getPrimaryKeyMatchCondition(selectX1, selectY1, primaryKeys.toArray(new String[0]));
-        Condition x1FromLessThanY1From = LessThan.of(sourceValidDatetimeFrom.withDatasetRef(selectX1.datasetReference()), sourceValidDatetimeFrom.withDatasetRef(selectY1.datasetReference()));
+        Condition x1FromLessThanY1From = LessThan.of(FieldValue.builder().datasetRef(selectX1.datasetReference()).fieldName(VALID_DATE_TIME_FROM_NAME).build(), FieldValue.builder().datasetRef(selectY1.datasetReference()).fieldName(VALID_DATE_TIME_FROM_NAME).build());
```
Comment (prasar-ashutosh) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java]: Can we define this FieldValue initally and then use withDatasetRef every where.
Then we will not need to define the Fielvalue.builder... every where

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java [hunk 3]
```diff
@@ -436,77 +447,73 @@ private Insert getStageToTemp()
...
         Join joinXY1 = Join.of(selectX1, selectY1, joinXY1Condition, JoinOperation.LEFT_OUTER_JOIN);
 
-        List<Value> selectXY1Fields = new ArrayList<>();
-        selectXY1Fields.addAll(primaryKeyFields.stream().map(field -> field.withDatasetRef(selectX1.datasetReference())).collect(Collectors.toList()));
-        selectXY1Fields.add(sourceValidDatetimeFrom.withDatasetRef(selectX1.datasetReference()));
+        List<Value> selectXY1Fields = primaryKeyFieldsAndFromFieldForSelection.stream().map(field -> field.withDatasetRef(selectX1.datasetReference())).collect(Collectors.toList());
         selectXY1Fields.add(FunctionImpl.builder()
             .functionName(FunctionName.COALESCE)
-            .addValue(FunctionImpl.builder().functionName(FunctionName.MIN).addValue(sourceValidDatetimeFrom.withDatasetRef(selectY1.datasetReference())).build(), LogicalPlanUtils.INFINITE_BATCH_TIME())
+            .addValue(FunctionImpl.builder().functionName(FunctionName.MIN).addValue(FieldValue.builder().datasetRef(selectY1.datasetReference()).fieldName(VALID_DATE_TIME_FROM_NAME).build()).build(), LogicalPlanUtils.INFINITE_BATCH_TIME())
```
Comment (prasar-ashutosh) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java]: Same comment as above, define a FieldValue and use it withDatasetRef every where

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java (hunk 1 above)
Comment (kumuwu) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java]: Sure changed!

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java (hunk 2 above)
Comment (kumuwu) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java]: Sure changed all places!

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java (hunk 3 above)
Comment (kumuwu) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java]: Sure changed all places!

**Review by prasar-ashutosh — APPROVED** [review]

**Pull request closed without merging**